cmd/note_common.go

```diff
@@ -166,42 +165,64 @@ func createCommitComments(project string, mrID int, commit string, body string,
 			}
 		} else if strings.HasPrefix(scanner.Text(), "|") {
 			if comments != "" {
-				createCommitNote(project, mrID, commit, newfile, oldfile, oldLineNum, newLineNum, comments, block)
+				createCommitNote(project, mrID, commit, newfile, oldfile, linetype, oldLineNum, newLineNum, comments, block)
```

Comment:
The fact that we duplicate this means it may be better to restructure this loop. Currently we have:

```go
for scanner.Scan() {
	if strings.HasPrefix(scanner.Text(), "| newfile:") {
		createCommitNote(project, mrID, commit, newfile, oldfile, linetype, oldLineNum, newLineNum, comments, block)
		// short body
	} else if strings.HasPrefix(scanner.Text(), "|") {
		createCommitNote(project, mrID, commit, newfile, oldfile, linetype, oldLineNum, newLineNum, comments, block)
		// long body
	} else {
		comments += "\n" + scanner.Text() // very short body
	}
}
```

I'd rather read this
(extracting scanner.Text() because it allocates on every call)

```go
for scanner.Scan() {
	line := scanner.Text()
	if !strings.HasPrefix(line, "|") {
		comments += "\n" + line
		continue
	}

	createCommitNote(project, mrID, commit, newfile, oldfile, linetype, oldLineNum, newLineNum, comments, block)
	
	if strings.HasPrefix(line, "| newfile:") {
		// short body
		continue
	}

	// long body
}
```

**However** since commenting on the very last context line of the last file in a diff doesn't work because there is no following "|" line, we should probably restructure this to just compute a list of comments, and submit all of them once we're done.

Reply:
Ah, good suggestion.  I've incorporated this change in an additional cleanup commit.